### DISCUSSION TITLE
Unable to open file on windows

### DISCUSSION THREAD

**Discussion by testanull** [github/securitylab#88] [General] [answered]
Hello everyone,
Recently, I got a strange problem.
With the same DB, it works normally on linux, but on windows i can not open file after click on the result
here is the debug log:
```Unable to handleMsgFromView: cannot open codeql-zip-archive://1-38/d%3A%5CResearch%5Csemmle%5CspringXdDB%5Csrc.zip/home/daphuc/Desktop/dbcollections/spring-xd/spring-xd-module-spi/src/main/java/org/springframework/xd/module/options/validation/DateWithCustomFormat.java. Detail: Unable to read file 'codeql-zip-archive://1-38/d:\Research\semmle\springXdDB\src.zip/home/daphuc/Desktop/dbcollections/spring-xd/spring-xd-module-spi/src/main/java/org/springframework/xd/module/options/validation/DateWithCustomFormat.java' (EntryNotFound (FileSystemError): codeql-zip-archive://1-38/d:\Research\semmle\springXdDB\src.zip/home/daphuc/Desktop/dbcollections/spring-xd/spring-xd-module-spi/src/main/java/org/springframework/xd/module/options/validation/DateWithCustomFormat.java)```
Have anyone got this problem before?

**Comment by jcreedcmu** [comment]
Are you experiencing this across linux and windows with the same query result? Or is it just that all result-links work on linux and none on windows?
Are you able to point us to the `.zip` file you're experiencing this with?

**Comment by jcreedcmu** [comment]
In particular, does your `d:\Research\semmle\springXdDB\src.zip` actually contain `home/daphuc/Desktop/dbcollections/spring-xd/spring-xd-module-spi/src/main/java/org/springframework/xd/module/options/validation/DateWithCustomFormat.java`, or a similarly named file?

**Comment by testanull** [comment]
I have tried on Mac, Linux and windows with the same DB, query and get error on windows
It's a normal file, i can still open by 7zip

**Comment by jcreedcmu** [comment, marked as answer]
And the file with the name I mentioned above definitely exists in it? Nothing funny going on with case or other special symbols or anything?

**Reply by testanull** [reply]
As I mentioned above, it exists and nothing too much special about it
Except the file path seems too long
Here is the DB, you can check it out: https://drive.google.com/file/d/1YURmLqY_lY6pY94wsiJ9izJPcdx2XZ3z/view?usp=sharing

**Reply by jcreedcmu** [reply]
Interesting --- what makes you think the path length is an issue? That does seem potentially plausible that that might have different consequences on different platforms.

**Reply by testanull** [reply]
Cause on windows there 's a limit on path length by 260 chars (https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file).
But I see that the current path is not long enough to be limit

**Reply by jcreedcmu** [reply]
When https://github.com/github/vscode-codeql/actions/runs/110396542 finishes building, you could try installing the `.vsix` that appears in the Artifacts tab, and the error message should have a bit more information about what the extension thinks it's trying to extract from the zip file. I wonder if there's some other platform-dependence in how the `unzipper` module works.

**Reply by testanull** [reply]
It's quite strange,
After installing your extension. now I'm able to open the code again.

**Reply by jcreedcmu** [reply]
Ok, glad you have a working extension again in any event. Maybe another bugfix along the way addressed whatever was going on. Definitely reopen if this (or any other bug) hits you again!

**Reply by testanull** [reply]
thank you so much for helping!
